### PR TITLE
Feature/eval framework

### PR DESCRIPTION
**Pull request by ATulenkov**
## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[x] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[x] No
```

## Type of change

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [ ] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [ ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Comment by ATulenkov** [comment]
@Pengjia-Aware @ratkinsoncinz just created PR for further suggestions

File: evaluation_framework/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    def prepare_tests(self, test_cases: List[Dict]) -> None:
+        """Run tests on the provided test cases"""
+        custom_metrics = read_json(self.custom_metrics)
+        metrics = create_metrics(self.llm_model, custom_metrics)
+        
+        test_cases = [LLMTestCase(**test_case) for test_case in test_cases]
+        return test_cases, metrics
+
+
+def create_llm_model() -> DeepEvalBaseLLM:
```
Comment (Pengjia-Aware) [inline evaluation_framework/evaluation.py]: Looks that these functions are only used inside class, perhaps moving into class can improve code organization

File: evaluation_framework/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            _answer = answers["results"][i]["message"]["content"]
+            _context = answers["results"][i]["context"]['data_points']["text"]
+            eval_case["actual_output"] = _answer
+            eval_case["retrieval_context"] = _context
+
+        return eval_data
+    
+    def prepare_goldens(self, goldens: List[Dict]) -> List[Dict]:
+        """Prepare goldens for evaluation"""
+        goldens = [g.model_dump() for g in goldens]
```
Comment (Pengjia-Aware) [inline evaluation_framework/evaluation.py]: can you just  return [g.model_dump() for g in goldens] ?

File: evaluation_framework/test_file.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    # Create dummy test cases (for expected behavior)
+    dummy_test_cases, metrics = evaluator.prepare_tests(test_data_dummy)
+    # Create RAG-answered test cases (actual system responses)
+    answered_test_data = await evaluator.prepare_eval_data(test_data_rag, RAG_system)
+    answered_test_cases, _ = evaluator.prepare_tests(answered_test_data)
+    
+    return dummy_test_cases, answered_test_cases, metrics
+
+# Run async setup once at module level
+dummy_test_cases, answered_test_cases, metrics = asyncio.run(run_evaluation())
```
Comment (Pengjia-Aware) [inline evaluation_framework/test_file.py:40]: You can move these codes to a main function

File: evaluation_framework/run_evaluation.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# Local imports
+from evaluation import RAGEvaluator
+from eval_config import EvalConfig
+from system_rag import RAG, create_RAG_eval_app, cleanup_clients
+from golden_generation import synthetaze_data
+
+
+logger = logging.getLogger(__name__)
+
+from dotenv import load_dotenv
```
Comment (Pengjia-Aware) [inline evaluation_framework/run_evaluation.py]: this is duplicated

File: evaluation_framework/run_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+            if len(eval_data) > 0:
+                await evaluator._evaluate(eval_data)
+                logger.info("Evaluation pipeline completed successfully")
+            else:
+                raise ValueError("No evaluation data to run")
+            
+        except Exception as e:
+            logger.error(f"Pipeline failed: {str(e)}")
+            raise ValueError(f"Pipeline failed: {str(e)}") from e
```
Comment (Pengjia-Aware) [inline evaluation_framework/run_evaluation.py]: raise may also log error, you may have two error log

File: evaluation_framework/run_evaluation.py (hunk 4 above)
Comment (ATulenkov) [reply evaluation_framework/run_evaluation.py]: Removed in a new commit

File: evaluation_framework/run_evaluation.py (hunk 5 above)
Comment (ATulenkov) [reply evaluation_framework/run_evaluation.py]: Fixed in a new commit

File: evaluation_framework/evaluation.py (hunk 1 above)
Comment (ATulenkov) [reply evaluation_framework/evaluation.py]: Fixed

**Comment by ATulenkov** [comment]
@Pengjia-Aware @ratkinsoncinz Added dashboard.py and updated requirements

**Pull request merged**